app/src/main/java/io/apicurio/registry/rest/ArtifactsResourceImpl.java

```diff
@@ -215,7 +222,7 @@ public Response getLatestArtifact(String artifactId) {
     public ArtifactMetaData updateArtifact(String artifactId, ArtifactType xRegistryArtifactType, InputStream data) {
         String content = toString(data);
         ArtifactType artifactType = determineArtifactType(content, xRegistryArtifactType, request);
-        // TODO -- apply rules
+        rulesService.applyRules(artifactId, artifactType, content, RuleApplicationType.update);
```

Comment:
One thing to consider - Is it possible that the execution of the rules may take too much time in some cases? In openshift for example, I think there is a time limit for service to respond to a request. We may have to do do this asynchronously later.

Reply:
If this happens to be the case - long execution - this then becomes a whole new problem, on how to best handle this.